spec/models/attachment_spec.rb

```diff
@@ -3,59 +3,62 @@
 
 RSpec.describe Attachment do
   let(:file_path) { File.join(Rails.root, '/spec/fixtures/files/text.txt') }
-  subject { build(:attachment, file: file_path) }
 
-  it { is_expected.to belong_to(:attachable) }
   it { is_expected.to respond_to(:url) }
   it { is_expected.to respond_to(:path) }
 
-  describe '#open' do
-    context 'when a block is provided' do
-      it 'yields a stream' do
-        subject.open do |file|
-          expect(file).to be_a(IO)
+  let!(:instance) { create(:instance) }
+  with_tenant(:instance) do
+    subject { build(:attachment_reference, file: file_path).attachment }
```

Comment:
I didn't make attachment a factory because we can't create files. otherwise the one to one mapping will link the all the created attachments to the same file. 

Or how about I just use a random hash in the factory ? regardless of the file hash.